Update examples with `@node` directive (#188)
neo4j/docs-graphql: modules/ROOT/pages/directives/database-mapping.adoc

Each type in your GraphQL type definitions can be mapped to an entity in your Neo4j database, such as nodes, relationships, and relationship properties.

[[type-definitions-node]]
== `@node`

Adding the `@node` directive to your GraphQL type specifies that it represents a Neo4j node.
For example, to represent a Neo4j node with the label "Movie" and a single property "title" of type string:

[source, graphql, indent=0]
----
type Movie @node {
    title: String
}
----

[NOTE]
====

> In the current version, it's not required to specify every GraphQL type representing a Neo4j node with the `@node` directive.

Reviewer: ```suggestion
In version 6.x, it's not required to specify every GraphQL type representing a Neo4j node with the `@node` directive.
```

> In the future, types without the `@node` directive will no longer treated as Neo4j nodes.

Reviewer: ```suggestion
In the future, types without the `@node` directive will no longer be treated as Neo4j nodes.
```

});
----

== `@relationship`

Relationships are represented by marking particular fields with a directive -- in this case, `@relationship`. 

> It defines the relationship type in the database, as well as it's direction.

Reviewer: ```suggestion
It defines the relationship type in the database, as well as its direction.
```

my own typo ;)

> It defines the relationship type in the database, as well as which direction that relationship goes in.

Reviewer: ```suggestion
It defines the relationship type in the database, as well as it's direction.
```

> To add a second node type, "Actor", and connect the two together, you should do the following:

Reviewer: ```suggestion
The following type definitions add a second node type, `Actor`, and connect `Movie` and `Actor` via the `ACTED_IN` relationship:
```

[source, graphql, indent=0]
----
type Movie @node {
    title: String
    actors: [Actor!]! @relationship(type: "ACTED_IN", direction: IN)
}

type Actor @node {
    name: String
    movies: [Movie!]! @relationship(type: "ACTED_IN", direction: OUT)
}
----

> Note that, in this case, there is a directive on each "end" of the relationship, but it is not essential.

Reviewer: ```suggestion
[NOTE]
====
The `@relationship` directive is used twice, once on each end of the relationship.
This is the standard way of modeling a relationship with the GraphQL library.
However, it is not a requirement of the type definitions themselves and relationships can deliberately be underspecified, for example to limit access through the API layer.
====
```

Please double-check that it's correct(ly phrased) what i stated

Author: Let's keep this outside this PR as it is as it is on 6.x, the only reason why it is marked as a changed is because I moved the `@node` description at the top of the page to remark the new importance.
